Bazeldeps: get lib file path by basename

kkpattern opened the pull request:
Changelog: Fix: BazelDeps generator cannot find a lib when it's named with the basename of the lib file.
Docs: omit

Users may not name their libraries in a conventional way.
Sometimes directly use the basename of the lib file as lib name.
We need to be able to find the lib in this case.

Close: #11331 

- [x] Refer to the issue that supports this Pull Request.
- [x] I've read the [Contributing guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've followed the PEP8 style guides for Python code.

<sup>**Note:** By default this PR will skip the slower tests and will use a limited set of python versions. Check [here](https://github.com/conan-io/conan/blob/develop/.github/PR_INCREASE_TESTING.md) how to increase the testing level by writing some tags in the current PR body text.</sup>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/conan-io/conan?pullRequest=11343) <br/>All committers have signed the CLA.

kkpattern committed:
Bazeldeps: get lib file path by basename

Users may not name their libraries in a conventional way.
Sometimes directly use the basename of the lib file as lib name.
We need to be able to find the lib in this case.

memsharded reviewed (approved):
Thanks very much for contributing this!
